airbyte-scheduler/src/main/java/io/airbyte/scheduler/persistence/DefaultJobCreator.java

```diff
@@ -0,0 +1,166 @@
+        .withCheckConnection(jobCheckConnectionConfig);
+
+    return jobPersistence.createJob(scope, jobConfig);
+  }
+
+  @Override
+  public long createDestinationCheckConnectionJob(DestinationConnection destination, String dockerImageName) throws IOException {
+    final String scope =
+        ScopeHelper.createScope(
+            JobConfig.ConfigType.CHECK_CONNECTION_DESTINATION,
+            destination.getDestinationId().toString());
+
+    final JobCheckConnectionConfig jobCheckConnectionConfig = new JobCheckConnectionConfig()
+        .withConnectionConfiguration(destination.getConfiguration())
+        .withDockerImage(dockerImageName);
+
+    final JobConfig jobConfig = new JobConfig()
+        .withConfigType(JobConfig.ConfigType.CHECK_CONNECTION_DESTINATION)
+        .withCheckConnection(jobCheckConnectionConfig);
+
+    return jobPersistence.createJob(scope, jobConfig);
+  }
+
+  @Override
+  public long createDiscoverSchemaJob(SourceConnection source, String dockerImageName) throws IOException {
+    final String scope = ScopeHelper.createScope(
+        JobConfig.ConfigType.DISCOVER_SCHEMA,
+        source.getSourceId().toString());
+
+    final JobDiscoverCatalogConfig jobDiscoverCatalogConfig = new JobDiscoverCatalogConfig()
+        .withConnectionConfiguration(source.getConfiguration())
+        .withDockerImage(dockerImageName);
+
+    final JobConfig jobConfig = new JobConfig()
+        .withConfigType(JobConfig.ConfigType.DISCOVER_SCHEMA)
+        .withDiscoverCatalog(jobDiscoverCatalogConfig);
+
+    return jobPersistence.createJob(scope, jobConfig);
+  }
+
+  @Override
+  public long createGetSpecJob(String integrationImage) throws IOException {
+    final String scope = ScopeHelper.createScope(
+        JobConfig.ConfigType.GET_SPEC,
+        integrationImage);
+
+    final JobConfig jobConfig = new JobConfig()
+        .withConfigType(JobConfig.ConfigType.GET_SPEC)
+        .withGetSpec(new JobGetSpecConfig().withDockerImage(integrationImage));
+
+    return jobPersistence.createJob(scope, jobConfig);
+  }
+
+  @Override
+  public long createSyncJob(SourceConnection source,
+                            DestinationConnection destination,
+                            StandardSync standardSync,
+                            String sourceDockerImageName,
+                            String destinationDockerImageName)
+      throws IOException {
+    final UUID connectionId = standardSync.getConnectionId();
+
+    final String scope = ScopeHelper.createScope(JobConfig.ConfigType.SYNC, connectionId.toString());
+
+    final JobSyncConfig jobSyncConfig = new JobSyncConfig()
+        .withSourceConnection(source)
+        .withSourceDockerImage(sourceDockerImageName)
+        .withDestinationConnection(destination)
+        .withDestinationDockerImage(destinationDockerImageName)
+        .withStandardSync(standardSync);
+
+    // todo (cgardens) - this will not have the intended behavior if the last job failed. then the next
```

Comment:
issue: https://github.com/airbytehq/airbyte/issues/1154. will fix this as my next task.